### ISSUE TITLE
Documents with no level-1 heading subsume all level-2 headings under the first level-2 heading

### ISSUE THREAD

**Issue by DivineDominion** [jrblevin/markdown-mode#571]
Sample document

```md
---
title: Wat
---
## Section a

### Sub 1

### Sub 11

## Section b

### Sub 2
```

The imenu output of `counsel-imenu` and also the outline in `treemacs` shows this hierarchy:

![image](https://user-images.githubusercontent.com/59080/102461312-7e8e2580-4048-11eb-894d-6acffa8d1f38.png)

I expected `Section b` and `Section a` to be siblings, but `Section b` is demoted to a child of `Section a`.

The code that is responsible for this is likely this parser:

https://github.com/jrblevin/markdown-mode/blob/4d7f525eed69aa98f50511cf3e834f27fce4c80e/markdown-mode.el#L5532-L5587

I may or may not be able to adjust this myself, we'll see, as my elisp isn't that good :) So I'm posting this issue to track the behavior. If someone else chimes in, great!.

**Comment by DivineDominion** [comment]
If all headings are level-2 headings, the nested display won't occur. In the parser, level-1 and level-2 are handled separately. 

So the level-3 match starting at line 5555 affects the result.

**Comment by syohex** [comment]
It looks markdown imenu function assumes there is at least one level1 header when there is a level2 header. I suppose the imenu function should be re-written.

**Comment by syohex** [comment]
@DivineDominion I have merged #574. Please check latest version

**Comment by DivineDominion** [comment]
That was super quick and helpful, thank you a thousandfold @syohex! 🎉
